python_modules/dagster/dagster/_config/pythonic_config/resource.py

```diff
@@ -862,21 +864,39 @@ def _is_annotated_as_resource_type(annotation: Type) -> bool:
     )
 
 
+class ResourceDataWithAnnotation(NamedTuple):
+    key: str
+    value: Any
+    annotation: Type
+    annotation_metadata: List[str]
+
+
 def separate_resource_params(cls: Type[BaseModel], data: Dict[str, Any]) -> SeparatedResourceParams:
     """Separates out the key/value inputs of fields in a structured config Resource class which
     are marked as resources (ie, using ResourceDependency) from those which are not.
     """
     keys_by_alias = {field.alias: field for field in cls.__fields__.values()}
-    data_with_annotation: List[Tuple[str, Any, Type]] = [
+    data_with_annotation: List[ResourceDataWithAnnotation] = [
         # No longer exists in Pydantic 2.x, will need to be updated when we upgrade
-        (k, v, keys_by_alias[k].outer_type_)
-        for k, v in data.items()
-        if k in keys_by_alias
+        ResourceDataWithAnnotation(
+            key=field_key,
+            value=field_value,
+            annotation=keys_by_alias[field_key].outer_type_,
+            annotation_metadata=[],
+        )
+        for field_key, field_value in data.items()
+        if field_key in keys_by_alias
     ]
+    # We need to grab metadata from the annotation in order to tell if
+    # this key was annotated with a typing.Annotated annotation (which we use for resource/resource deps),
+    # since Pydantic 2.0 strips that info out and sticks any Annotated metadata in the
+    # metadata field
     out = SeparatedResourceParams(
-        resources={k: v for k, v, t in data_with_annotation if _is_annotated_as_resource_type(t)},
+        resources={
+            k: v for k, v, t, m in data_with_annotation if _is_annotated_as_resource_type(t, m)
+        },
         non_resources={
-            k: v for k, v, t in data_with_annotation if not _is_annotated_as_resource_type(t)
+            k: v for k, v, t, m in data_with_annotation if not _is_annotated_as_resource_type(t, m)
```

Comment:
please use the properties here. otherwise they are not worth introducing. i.e.

```python
resources={
   d.key : d.value for d in d in data_with_annotation if _is_annotated_as_resource_type(d.annotation, d.annotation_with_metadata)
    },
 },
```